core/src/main/scala/org/apache/spark/scheduler/SchedulableBuilder.scala

```diff
@@ -69,60 +72,81 @@ private[spark] class FairSchedulableBuilder(val rootPool: Pool, conf: SparkConf)
   val DEFAULT_WEIGHT = 1
 
   override def buildPools() {
-    var is: Option[InputStream] = None
+    var fileData: Option[FileData] = None
     try {
-      is = Option {
-        schedulerAllocFile.map { f =>
-          new FileInputStream(f)
-        }.getOrElse {
-          Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
-        }
-      }
-
-      is.foreach { i => buildFairSchedulerPool(i) }
+      fileData = getFileData()
+      fileData.foreach { data => buildFairSchedulerPool(data) }
+    } catch {
+      case NonFatal(t) =>
+        logError("Error while building the fair scheduler pools: ", t)
+        throw t
     } finally {
-      is.foreach(_.close())
+      fileData.foreach(_.inputStream.close())
     }
 
     // finally create "default" pool
     buildDefaultPool()
   }
 
+  private def getFileData(): Option[FileData] = {
+    schedulerAllocFile.map { f =>
+      val file = new File(f)
+      val fis = new FileInputStream(file)
+      logInfo(s"Creating Fair Scheduler pools from ${file.getName}")
+      Some(FileData(fis, file.getName))
+    }.getOrElse {
+      val is = Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
+      if(is != null) {
+        logInfo(s"Creating Fair Scheduler pools from default file: $DEFAULT_SCHEDULER_FILE")
+        Some(FileData(is, DEFAULT_SCHEDULER_FILE))
+      }
+      else {
+        logWarning("Fair Scheduler configuration file not found so jobs will be scheduled " +
+          "in FIFO order")
+        None
+      }
+    }
+  }
+
   private def buildDefaultPool() {
     if (rootPool.getSchedulableByName(DEFAULT_POOL_NAME) == null) {
       val pool = new Pool(DEFAULT_POOL_NAME, DEFAULT_SCHEDULING_MODE,
         DEFAULT_MINIMUM_SHARE, DEFAULT_WEIGHT)
       rootPool.addSchedulable(pool)
-      logInfo("Created default pool %s, schedulingMode: %s, minShare: %d, weight: %d".format(
+      logInfo("Created default pool: %s, schedulingMode: %s, minShare: %d, weight: %d".format(
         DEFAULT_POOL_NAME, DEFAULT_SCHEDULING_MODE, DEFAULT_MINIMUM_SHARE, DEFAULT_WEIGHT))
     }
   }
 
-  private def buildFairSchedulerPool(is: InputStream) {
-    val xml = XML.load(is)
+  private def buildFairSchedulerPool(fileData: FileData) {
```

Comment:
it would be better to make this method unaware of the FileData class, and instead just have two input params: the InputStream and the Filename